quix-frontend/client/src/lib/sql-autocomplete/adapter/sql-autocomplete-adapter.ts

```diff
@@ -47,6 +48,25 @@ export class SqlAutocompleter implements IAutocompleter {
         return [];
     }
   }
+  public async getAllCompletionItemsFromQueryContextCollumn(queryContext: any) {
+    const { contextType } = queryContext;
+    const complitionsArray = contextType === ContextType.Column ? this.translateAndGetAllQueryContextColumns(queryContext.tables ,queryContext.prefix) :[] ;
+    return complitionsArray
+  }
+
+  public async translateAndGetAllQueryContextColumns(tables: TableInfo[] , prefix: string | undefined) {
+    const tablesPromises: Promise<TableInfo>[] = [];
+    for (const table of tables) {
+      tablesPromises.push(this.extractTableColumns(table));
+    }
+     const extractedTables = await Promise.all(tablesPromises);
```

Comment:
I see that exact same logic exists in another function.. please extract it to a function and use it in both of them..